File: pojtotailo.js
// Reference: https://en.wiktionary.org/wiki/Module:nan-pron
// using function export.poj_to_tl_conv(text)

function poj_to_tailo(text) {
    var accent = '[́̀̂]';
    var conv = {
        "e" :  "i",
        "E" : "I",
        "o" : "u",
        "O" : "U"
    };

    function convert(match, p1, p2, offset, string) {
        return conv[p1] + p2;
    }

    text = text.replace('#', '');

    text = text.normalize('NFD');

    text = text.replace('\u4ED4', '\u00E1');

    text = text.replace(new RegExp('\\(([^\\)]+)\\)'), '$1-$1-$1');

    text = text.replace(new RegExp('([eE])(' + '\\u030D' + '?k)'), convert);

    text = text.replace(new RegExp('([eE])(' + accent + '?ng)'), convert);

    text = text.replace(new RegExp('([oO])(' + accent + '?[ae])'), convert);

    text = text.replace(new RegExp('([uU])(' + accent + '?)([aei])'), '$1$3$2');

    text = text.replace(new RegExp('([eE])(' + accent + '?)(re)'), '$1$3$2');

    text = text.replace(new RegExp('([oO]' + accent + '?)\\u0358'), '$1o');

    text = text.replace(new RegExp('(h?)\\u207F'), 'nn$1');

    text = text.replace(new RegExp('([cC]h)'), (match, p1, offset, string) => {
        if (match == "ch") {
            return "ts";
        } else if (match == "Ch") {
            return "Ts"
        }
    });

    text = text.replace(new RegExp('\\/([^\\s])'), ' / $1');

    text = text.normalize('NFC');
    console.log(RegExp('\\(([^\\)]+)\\)'))
    return text
}

File: pojtotailo.lua
function export.poj_to_tl_conv(text)
    if type(text) == "table" then text = text.args[1] end
    local accent = "[́̀̂]?"
    local conv = {
        ["e"] = "i", ["E"] = "I", ["o"] = "u", ["O"] = "U"
    }
    local function convert(a, b)
        return conv[a] .. b
    end
    text = gsub(text, "#", "")
    text = mw.ustring.toNFD(text)
    text = gsub(text, "仔", "á")
    text = gsub(text, "%(([^%)]+)%)", "%1-%1-%1")
    text = gsub(text, "([eE])(̍?k)", convert)
    text = gsub(text, "([eE])(" .. accent .. "ng)", convert)
    text = gsub(text, "([oO])(" .. accent .. "[ae])", convert)
    text = gsub(text, "([uU])(" .. accent .. ")([aei])", "%1%3%2")
    text = gsub(text, "([eE])(" .. accent .. ")(re)", "%1%3%2")
    text = gsub(text, "([oO]" .. accent .. ")͘", "%1o")
    text = gsub(text, "(h?)ⁿ", "nn%1")
    text = gsub(text, "[cC]h", {["ch"] = "ts", ["Ch"] = "Ts"})
    text = gsub(text,'/([^ ])',' / %1')
    return mw.ustring.toNFC(text)
end